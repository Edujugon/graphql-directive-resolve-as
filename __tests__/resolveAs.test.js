const { graphql } = require('graphql');
const nock = require('nock');
const { resolveAs } = require('../src/index');
const schema = require('../example/schema');

beforeAll(() => {
  nock.disableNetConnect();
});

afterEach(() => {
  nock.cleanAll();
});

afterAll(() => {
  nock.enableNetConnect();
});

test('getDirectiveDeclaration should be defined', () => {
  expect(resolveAs.getDirectiveDeclaration()).toMatchSnapshot();
});

test('resolves correctly', async () => {
  const response = await graphql(
    schema,
    `
      query {
        me {
          name
          lastName
        }
      }
    `
  );
  expect(response).toMatchSnapshot();
});

test('response is expected object', async () => {
  const response = await graphql(
    schema,
    `
      query {
        me {
          name,
          country,
          city,
          neighborhood,
          firstFriend
        }
      }
    `
  );
  expect(response).toEqual(
    {"data": {"me": {"name": "John", "country": "Germany", "city": "Berlin", "neighborhood": "Kreuzberg", "firstFriend": "Edu"}}}
  );
});
