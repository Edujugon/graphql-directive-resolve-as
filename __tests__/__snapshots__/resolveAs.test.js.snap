// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDirectiveDeclaration should be defined 1`] = `
GraphQLDirective {
  "args": Array [
    Object {
      "astNode": undefined,
      "defaultValue": undefined,
      "description": null,
      "name": "name",
      "type": "String",
    },
  ],
  "astNode": undefined,
  "description": undefined,
  "locations": Array [
    "FIELD_DEFINITION",
  ],
  "name": "resolveAs",
}
`;

exports[`resolves correctly 1`] = `
Object {
  "data": Object {
    "me": Object {
      "lastName": "Doe",
      "name": "John",
    },
  },
}
`;
